pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,477 @@
+import re
+
+import numpy as np
+
+from pyfr.ctypesutil import LibWrapper
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin
+from pyfr.shapes import BaseShape
+from pyfr.util import file_path_gen, subclass_where
+from pyfr.writers.vtk import BaseShapeSubDiv
+
+
+class ConduitWrappers(LibWrapper):
+    _libname = 'conduit'
+    _mode = RTLD_GLOBAL
+
+    # Functions
+    _functions = [
+        (c_void_p, 'conduit_node_append', c_void_p),
+        (c_void_p, 'conduit_node_create', c_void_p),
+        (None, 'conduit_node_destroy', c_void_p),
+        (c_void_p, 'conduit_node_fetch', c_void_p, c_char_p),
+        (c_void_p, 'conduit_node_set_path_char8_str', c_void_p, c_char_p,
+         c_char_p),
+        (c_void_p, 'conduit_node_set_path_float32', c_void_p, c_char_p,
+         c_float),
+        (c_void_p, 'conduit_node_set_path_float32_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_float64', c_void_p, c_char_p, 
+         c_double),
+        (c_void_p, 'conduit_node_set_path_float64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_int32', c_void_p, c_char_p, c_long),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p),
+        (c_void_p, 'conduit_node_print', c_void_p),
+        (c_void_p, 'conduit_node_print_detailed', c_void_p),
+    ]
+
+
+class ConduitNode:
+    def __init__(self, lib, ptr=None, child=False):
+        self.lib = lib
+        self.child = child
+
+        if ptr is not None:
+            self._as_parameter_ = ptr
+        else:
+            self._as_parameter_ = self.lib.conduit_node_create(None)
+
+    def __del__(self):
+        ptr = self._as_parameter_
+        if hasattr(self, 'lib') and ptr is not None and not self.child:
+            self.lib.conduit_node_destroy(self)
+        self._as_parameter_ = None
+
+    def __getitem__(self, key):
+        ptr = c_void_p(self.lib.conduit_node_fetch(self, key.encode()))
+        return ConduitNode(lib=self.lib, ptr=ptr)
+
+    def __setitem__(self, key, value):
+        match value:
+            case str():
+                self.lib.conduit_node_set_path_char8_str(self, key.encode(),
+                    value.encode())
+            case ConduitNode():
+                self.lib.conduit_node_set_path_node(self, key.encode(),
+                    value)
+            case int():
+                self.lib.conduit_node_set_path_int32(self, key.encode(),
+                    c_long(value))
+            case float():
+                self.lib.conduit_node_set_path_float32(self, key.encode(),
+                    c_float(value))
+            case (np.ndarray(dtype='float32') | np.generic(dtype='float32')):
+                vc = np.ascontiguousarray(value)
```

Comment:
This code is still way too complicated.

Reply:
What do you suggest?

Reply:
```
vc = np.ascontiguousarray(value)
self.lib.conduit_node_set_path_float32_ptr(self, key.encode(), vc.ctypes.data, vc.size)
```
Although you can make life easier by doing `key = key.encode()` at the top.  A nested match/case or if chain may make life easier since the only thing which changes between different array types is the function name (condut...).